File: 2204-find-subsequence-of-length-k-with-the-largest-sum/2204-find-subsequence-of-length-k-with-the-largest-sum.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        vector<int> v = nums;
        sort(v.begin(),v.end(),greater<int>());
        vector<int> ans;
        unordered_map<int,int> cnt;  
        for(int i = 0;i<k;i++){
            cnt[v[i]]++;
        }
        for(auto &num : nums){
            if(cnt[num]){
                ans.push_back(num);
                cnt[num]--;
            }
        }
        return ans;
    }
};

File: 2204-find-subsequence-of-length-k-with-the-largest-sum/2204-find-subsequence-of-length-k-with-the-largest-sum.cs
public class Solution {
    public int[] MaxSubsequence(int[] nums, int k) {
        int[] arr = nums.OrderByDescending(x => x).ToArray();
        Dictionary<int,int> cnt = new Dictionary<int,int>();
        for(int i = 0;i<k;i++){
            if (!cnt.ContainsKey(arr[i]))
                cnt[arr[i]] = 0;
            cnt[arr[i]]++;
        } 
        List<int> ans = new List<int>();
        foreach(int num in nums){
            if(cnt.ContainsKey(num) && cnt[num] > 0){
                ans.Add(num);
                cnt[num]--;
            }
        }
        return ans.ToArray();
    }
}